From db20f1c868e5da83b1bebda67d98b590f3eccb80 Mon Sep 17 00:00:00 2001
From: Stephen Arnold <sarnold@vctlabs.com>
Date: Fri, 9 Dec 2016 18:53:10 -0800
Subject: [PATCH 1/2] dts: socfpga_cyclone5_de1_soc: add 2 new de1_soc files
 for FB and audio

* update audio dts (from DE1-SoC-Sound project) to latest

Signed-off-by: Stephen Arnold <sarnold@vctlabs.com>
---
 arch/arm/boot/dts/Makefile                         |   2 +
 .../boot/dts/socfpga_cyclone5_de1_soc-audio.dts    | 271 +++++++++++++++++++++
 arch/arm/boot/dts/socfpga_cyclone5_de1_soc-fb.dts  | 169 +++++++++++++
 3 files changed, 442 insertions(+)
 create mode 100644 arch/arm/boot/dts/socfpga_cyclone5_de1_soc-audio.dts
 create mode 100644 arch/arm/boot/dts/socfpga_cyclone5_de1_soc-fb.dts

diff --git a/arch/arm/boot/dts/Makefile b/arch/arm/boot/dts/Makefile
index 1b5b294f66f2..b7163d737ea4 100644
--- a/arch/arm/boot/dts/Makefile
+++ b/arch/arm/boot/dts/Makefile
@@ -702,6 +702,8 @@ dtb-$(CONFIG_ARCH_SOCFPGA) += \
 	socfpga_cyclone5_mcvevk.dtb \
 	socfpga_cyclone5_socdk.dtb \
 	socfpga_cyclone5_de0_sockit.dtb \
+	socfpga_cyclone5_de1_soc-audio.dtb \
+	socfpga_cyclone5_de1_soc-fb.dtb \
 	socfpga_cyclone5_sockit.dtb \
 	socfpga_cyclone5_socrates.dtb \
 	socfpga_cyclone5_trcom.dtb \
diff --git a/arch/arm/boot/dts/socfpga_cyclone5_de1_soc-audio.dts b/arch/arm/boot/dts/socfpga_cyclone5_de1_soc-audio.dts
new file mode 100644
index 000000000000..dfbb7a550d43
--- /dev/null
+++ b/arch/arm/boot/dts/socfpga_cyclone5_de1_soc-audio.dts
@@ -0,0 +1,271 @@
+/*
+ *  Copyright (C) 2013 Steffen Trumtrar <s.trumtrar@pengutronix.de>
+ *  Copyright (C) 2016 Stephen Arnold <nerdboy@gentoo.org>
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "socfpga_cyclone5.dtsi"
+#include <dt-bindings/interrupt-controller/irq.h>
+
+/ {
+	model = "Terasic DE1-SoC";
+	compatible = "altr,socfpga-cyclone5", "altr,socfpga";
+
+	chosen {
+		bootargs = "earlyprintk";
+		stdout-path = "serial0:115200n8";
+	};
+
+	memory {
+		name = "memory";
+		device_type = "memory";
+		reg = <0x0 0x40000000>; /* 1GB */
+	};
+
+	aliases {
+		ethernet0 = &gmac1;
+	};
+
+	regulator_3_3v: 3-3-v-regulator {
+		compatible = "regulator-fixed";
+		regulator-name = "3.3V";
+		regulator-min-microvolt = <3300000>;
+		regulator-max-microvolt = <3300000>;
+	};
+
+	leds {
+		compatible = "gpio-leds";
+		user {
+			label = "user-led";
+			gpios = <&portb 24 0>;
+			linux,default-trigger = "heartbeat";
+		};
+#if 0
+		/* Always-on to enable HPS i2c */
+		/* The alternative is to model it as a regulator for codec */
+		i2c0mux {
+			label = "i2c0-mux";
+			gpios = <&portb 19 0>;
+			linux,default-trigger = "default-on";
+		};
+#endif
+	};
+
+        gpio_keys {
+		compatible = "gpio-keys";
+		#address-cells = <1>;
+		#size-cells = <0>;
+
+		user_pb {
+			label = "user_pb";
+			gpios = <&portb 25 0>;
+			linux,code = <28>;
+			gpio-key,wakeup;
+                };
+        };
+
+        sound {
+	        compatible = "opencores,de1soc-wm8731-audio";
+		i2s-controller = <&i2s>;
+		audio-codec = <&wm8731>;
+		//i2c-mux-gpio = <&portb 19 0>;
+        };
+
+	clk48: clk48 {
+		compatible = "fixed-clock";
+		#clock-cells = <0>;
+		clock-frequency  = <24576000>;
+		clock-output-names = "clk48";
+	};
+
+	clk44: clk44 {
+		compatible = "fixed-clock";
+		#clock-cells = <0>;
+		clock-frequency  = <33868800>;
+		clock-output-names = "clk44";
+	};
+
+	i2s: i2s@0 {
+		#sound-dai-cells = <1>;
+		compatible = "opencores,i2s";
+		reg = <0xff200000 0x20>, <0xff200020 0x20>;
+		clocks = <&clk44>, <&clk48>;
+		clock-names = "clk44", "clk48";
+		dmas = <&pdma 0>, <&pdma 1>;
+		dma-names = "tx", "rx";
+	};
+#if 0
+	seg7@20 {
+		compatible = "dummy";
+		reg = <0xff200020 0x20>;
+	};
+#endif
+
+	vip2@100 {
+		compatible = "altr,vip-frame-reader-1.0", "altr,vip-frame-reader-9.1";
+		reg = <0xff200100 0x180>;
+		max-width = <0x400>;
+		max-height = <0x300>;
+		mem-word-width = <0x80>;
+		bits-per-color = <0x8>;
+	};
+
+	sysid@2000 {
+		compatible = "altr,sysid-1.0";
+		reg = <0xff202000 0x8>;
+	};
+#if 0
+	led_pio@10040 {
+		compatible = "altr,pio-1.0";
+		reg = <0xff210040 0x10>;
+		altr,gpio-bank-width = <10>;
+		#gpio-cells = <1>;
+		gpio-controller;
+	};
+	dipsw_pio@10080 {
+		compatible = "altr,pio-1.0";
+		reg = <0xff210080 0x10>;
+		interrupts = <0 42 4>;
+		altr,gpio-bank-width = <10>;
+		altr,interrupt_type = <IRQ_TYPE_EDGE_BOTH>;
+		#gpio-cells = <1>;
+		gpio-controller;
+		#interrupt-cells = <1>;
+		interrupt-controller;
+	};
+	button_pio@100c0 {
+		compatible = "altr,pio-1.0";
+		reg = <0xff2100c0 0x10>;
+		interrupts = <0 41 4>;
+		altr,gpio-bank-width = <4>;
+		altr,interrupt_type = <IRQ_TYPE_EDGE_FALLING>;
+		#gpio-cells = <1>;
+		gpio-controller;
+		#interrupt-cells = <1>;
+		interrupt-controller;
+	};
+	jtag_uart@20000 {
+		compatible = "altr,juart-1.0";
+		reg = <0xff220000 0x8>;
+		interrupts = <0 40 4>;
+	};
+	/*
+	fpgadma: fifo {
+		#address-cells = <1>;
+		#size-cells = <1>;
+		compatible = "altr,fpga-dma";
+		reg = <0xff200060 0x20>, <0xc0001000 0x10>;
+		reg-names = "csr", "data";
+		dmas = <&pdma 2 &pdma 3>;
+		dma-names = "tx", "rx";
+	};
+	*/
+#endif
+};
+
+&osc1 {
+	clock-frequency = <25000000>;
+};
+
+&gmac1 {
+	status = "okay";
+	phy-mode = "rgmii";
+
+	txd0-skew-ps = <0>; /* -420ps */
+	txd1-skew-ps = <0>; /* -420ps */
+	txd2-skew-ps = <0>; /* -420ps */
+	txd3-skew-ps = <0>; /* -420ps */
+	rxd0-skew-ps = <420>; /* 0ps */
+	rxd1-skew-ps = <420>; /* 0ps */
+	rxd2-skew-ps = <420>; /* 0ps */
+	rxd3-skew-ps = <420>; /* 0ps */
+	txen-skew-ps = <0>; /* -420ps */
+	txc-skew-ps = <1860>; /* 960ps */
+	rxdv-skew-ps = <420>; /* 0ps */
+	rxc-skew-ps = <1680>; /* 780ps */
+
+	max-frame-size = <3800>;
+};
+
+&gpio0 {
+	status = "okay";
+};
+
+&gpio1 {
+	status = "okay";
+};
+
+&gpio2 {
+	status = "okay";
+};
+
+&qspi {
+	status = "okay";
+};
+
+&spi0 {
+	status = "okay";
+
+	spidev@0 {
+		/* spidev causes a WARN_ON() so spoof with DAC compat */
+		compatible = "rohm,dh2228fv";
+		reg = <0>;      /* chip select */
+		spi-max-frequency = <1000000>;
+		enable-dma = <1>;
+	};
+};
+
+&uart0 {
+	status = "okay";
+};
+
+&usb1 {
+	status = "okay";
+};
+
+&spi1 {
+	status = "okay";
+};
+
+&mmc0 {
+	vmmc-supply = <&regulator_3_3v>;
+	vqmmc-supply = <&regulator_3_3v>;
+	status = "okay";
+};
+
+&i2c0 {
+	status = "okay";
+	//clock-frequency = <100000>;
+	speed-mode = <0>;
+
+	wm8731: wm8731@34 {
+		#sound-dai-cells = <0>;
+		compatible = "wlf,wm8731";
+		reg = <0x1a>;
+	};
+
+	adxl345: adxl345@0 {
+		compatible = "adi,adxl34x";
+		reg = <0x53>;
+
+		interrupt-parent = <&portc>;
+		interrupts = <3 2>;
+	};
+};
+
+&i2c1 {
+	status = "okay";
+	speed-mode = <0>;
+};
diff --git a/arch/arm/boot/dts/socfpga_cyclone5_de1_soc-fb.dts b/arch/arm/boot/dts/socfpga_cyclone5_de1_soc-fb.dts
new file mode 100644
index 000000000000..23701ad94c60
--- /dev/null
+++ b/arch/arm/boot/dts/socfpga_cyclone5_de1_soc-fb.dts
@@ -0,0 +1,169 @@
+/*
+ *  Copyright (C) 2016 Stephen Arnold <nerdboy@gentoo.org>
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "socfpga_cyclone5.dtsi"
+#include <dt-bindings/interrupt-controller/irq.h>
+
+/ {
+	model = "Terasic DE1-SoC";
+	compatible = "altr,socfpga-cyclone5", "altr,socfpga";
+
+	chosen {
+		bootargs = "earlyprintk";
+		stdout-path = "serial0:115200n8";
+	};
+
+	memory {
+		name = "memory";
+		device_type = "memory";
+		reg = <0x0 0x40000000>; /* 1GB */
+	};
+
+	aliases {
+		ethernet0 = &gmac1;
+	};
+
+	regulator_3_3v: 3-3-v-regulator {
+		compatible = "regulator-fixed";
+		regulator-name = "3.3V";
+		regulator-min-microvolt = <3300000>;
+		regulator-max-microvolt = <3300000>;
+	};
+
+	leds {
+		compatible = "gpio-leds";
+		user {
+			label = "user-led";
+			gpios = <&portb 24 0>;
+			linux,default-trigger = "heartbeat";
+		};
+	};
+
+        gpio_keys {
+		compatible = "gpio-keys";
+		#address-cells = <1>;
+		#size-cells = <0>;
+
+		user_pb {
+			label = "user_pb";
+			gpios = <&portb 25 0>;
+			linux,code = <28>;
+			gpio-key,wakeup;
+                };
+        };
+
+	alt_vip_vfr_0: vip@0xff260000 {
+		compatible = "altr,vip-frame-reader-1.0";
+		reg = <0xff260100 0x00000080>;
+		max-width = <1024>;
+		max-height = <768>;
+		mem-word-width = <128>;
+		bits-per-color = <8>;
+	};
+
+	sysid@2000 {
+		compatible = "altr,sysid-1.0";
+		reg = <0xff202000 0x8>;
+	};
+};
+
+&osc1 {
+	clock-frequency = <25000000>;
+};
+
+&gmac1 {
+	status = "okay";
+	phy-mode = "rgmii";
+
+	txd0-skew-ps = <0>; /* -420ps */
+	txd1-skew-ps = <0>; /* -420ps */
+	txd2-skew-ps = <0>; /* -420ps */
+	txd3-skew-ps = <0>; /* -420ps */
+	rxd0-skew-ps = <420>; /* 0ps */
+	rxd1-skew-ps = <420>; /* 0ps */
+	rxd2-skew-ps = <420>; /* 0ps */
+	rxd3-skew-ps = <420>; /* 0ps */
+	txen-skew-ps = <0>; /* -420ps */
+	txc-skew-ps = <1860>; /* 960ps */
+	rxdv-skew-ps = <420>; /* 0ps */
+	rxc-skew-ps = <1680>; /* 780ps */
+
+	max-frame-size = <3800>;
+};
+
+&gpio0 {
+	status = "okay";
+};
+
+&gpio1 {
+	status = "okay";
+};
+
+&gpio2 {
+	status = "okay";
+};
+
+&qspi {
+	status = "okay";
+};
+
+&spi0 {
+	status = "okay";
+
+	spidev@0 {
+		/* spidev causes a WARN_ON() so spoof with DAC compat */
+		compatible = "rohm,dh2228fv";
+		reg = <0>;      /* chip select */
+		spi-max-frequency = <1000000>;
+		enable-dma = <1>;
+	};
+};
+
+&i2c0 {
+	status = "okay";
+	speed-mode = <0>;
+
+	adxl345: adxl345@0 {
+		compatible = "adi,adxl34x";
+		reg = <0x53>;
+
+		interrupt-parent = <&portc>;
+		interrupts = <3 2>;
+	};
+};
+
+&mmc0 {
+	vmmc-supply = <&regulator_3_3v>;
+	vqmmc-supply = <&regulator_3_3v>;
+};
+
+&uart0 {
+	status = "okay";
+};
+
+&usb1 {
+	status = "okay";
+};
+
+&i2c1 {
+	status = "okay";
+	speed-mode = <0>;
+};
+
+&spi1 {
+	status = "okay";
+};
-- 
2.13.1

